src/main/scala/li/cil/oc/server/machine/luac/LuaStateFactory.scala

```diff
@@ -170,13 +173,26 @@ abstract class LuaStateFactory {
 
   def isAvailable = haveNativeLibrary
 
-  val is64Bit = Architecture.IS_OS_X64
+  val is64Bit = {
```

Comment:
Thank you for the effort, but...

This function in general is pretty silly. We use it to figure out whether or not to compensate for larger pointer sizes when calculating the maximum amount of available RAM - in practice, I should instead add a JNI method to simply *ask* what the Lua instance's pointer size is, and adjust accordingly.